[WPE] WPE Platform: add API to set user data to WPEBuffer

carlosgcampos opened the pull request:
#### 061466c843bb851a8a51aeb44c3674fe852fdfdc
<pre>
[WPE] WPE Platform: add API to set user data to WPEBuffer
<a href="https://bugs.webkit.org/show_bug.cgi?id=266701">https://bugs.webkit.org/show_bug.cgi?id=266701</a>

Reviewed by Michael Catanzaro.

It&apos;s common to associate a platform buffer to a WPEBuffer. We currently
use g_object_set_data_full for that, but it&apos;s more covenient with the
API provides a specific API for it.

* Source/WebKit/WPEPlatform/wpe/WPEBuffer.cpp:
(wpeBufferDispose):
(wpe_buffer_class_init):
(wpe_buffer_set_user_data):
(wpe_buffer_get_user_data):
* Source/WebKit/WPEPlatform/wpe/WPEBuffer.h:
* Source/WebKit/WPEPlatform/wpe/drm/WPEViewDRM.cpp:
(drmBufferCreateDMABuf):
(wpeViewDRMRequestUpdate):
(wpeViewDRMRenderBuffer):
* Source/WebKit/WPEPlatform/wpe/wayland/WPEViewWayland.cpp:
(createWaylandBufferFromDMABuf):
(createWaylandBufferSHM):

Canonical link: <a href="https://commits.webkit.org/272344@main">https://commits.webkit.org/272344@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/34b3306a0230195a4fe746e40c57bd52feada22a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/31407 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/10077 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/33102 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/33901 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/28460 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/12424 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/7325 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/28089 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/31744 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/8486 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/28040 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/7306 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/7463 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/35243 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/28550 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/28399 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/33601 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/7537 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/5566 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/31442 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/9200 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/8229 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/4089 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/8049 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro commented on Source/WebKit/WPEPlatform/wpe/WPEBuffer.cpp:
```diff
@@ -190,6 +203,45 @@ int wpe_buffer_get_height(WPEBuffer* buffer)
+ * wpe_buffer_set_user_data:
+ * @buffer: a #WPEBuffer
+ * @user_data: data to associate with @buffer
+ * @destroy_func: the function to call to release @user_data
```
```suggestion
 * @destroy_func: (nullable): the function to call to release @user_data
```

mcatanzaro reviewed (approved):
Certainly much nicer than using g_object_set_data_full(), which requires global locking and is not good for anything performance sensitive.

carlosgcampos committed:
[WPE] WPE Platform: add API to set user data to WPEBuffer
https://bugs.webkit.org/show_bug.cgi?id=266701

Reviewed by Michael Catanzaro.

It's common to associate a platform buffer to a WPEBuffer. We currently
use g_object_set_data_full for that, but it's more covenient with the
API provides a specific API for it.

* Source/WebKit/WPEPlatform/wpe/WPEBuffer.cpp:
(wpeBufferDispose):
(wpe_buffer_class_init):
(wpe_buffer_set_user_data):
(wpe_buffer_get_user_data):
* Source/WebKit/WPEPlatform/wpe/WPEBuffer.h:
* Source/WebKit/WPEPlatform/wpe/drm/WPEViewDRM.cpp:
(drmBufferCreateDMABuf):
(wpeViewDRMRequestUpdate):
(wpeViewDRMRenderBuffer):
* Source/WebKit/WPEPlatform/wpe/wayland/WPEViewWayland.cpp:
(createWaylandBufferFromDMABuf):
(createWaylandBufferSHM):

Canonical link: https://commits.webkit.org/272344@main
